### PR TITLE
Support structure logging for livenessprobe container

### LINKED ISSUE #165: Support structured logging
The `livenessprobe` container should support structured logging when given the `--log-format-json` flag, like for instance the `secrets-store-csi-driver` does.

A simple implementation can be found here:  https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/main/cmd/secrets-store-csi-driver/main.go#L90-L94

### PR DESCRIPTION
**Pull request by Kartik494**
**What type of PR is this?**
> /kind bug

**What this PR does / why we need it**:
This PR add support for structure logging on livenessprobe container 

**Which issue(s) this PR fixes**:

Fixes #165 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Added support for structure logging
```

### CODE REVIEW

**Comment by Kartik494** [comment]
@mauriciopoppe Could you please take a look and suggest if this  changes seems ok to you.

**Comment by Kartik494** [comment]
Added commit to use Suffix S `InfoS`.

File: cmd/livenessprobe/main.go [hunk 1]
```diff
@@ -131,6 +134,11 @@ func main() {
 		klog.Error("only one of `--metrics-address` and `--http-endpoint` can be explicitly set.")
 		os.Exit(1)
 	}
+	if *logFormatJSON {
+		jsonFactory := json.Factory{}
+		logger, _ := jsonFactory.Create(logsapi.LoggingConfiguration{Format: "json"})
+		klog.InfoS("logging livenessprobe",logger)
```
Comment (mauriciopoppe) [inline cmd/livenessprobe/main.go:140]: I think this is just logging `logger` but not telling klog to use it for logging

File: cmd/livenessprobe/main.go [hunk 2]
```diff
@@ -46,6 +48,7 @@ var (
 	metricsAddress = flag.String("metrics-address", "", "(deprecated) The TCP network address where the prometheus metrics endpoint will listen (example: `:8080`). The default is empty string, which means metrics endpoint is disabled. If set, `--http-endpoint` cannot be set, and the address cannot resolve to localhost + the port from `--health-port`.")
 	httpEndpoint   = flag.String("http-endpoint", "", "The TCP network address where the HTTP server for diagnostics, including CSI driver health check and metrics. The default is empty string, which means the server is disabled. If set, `--health-port` and `--metrics-address` cannot be explicitly set.")
 	metricsPath    = flag.String("metrics-path", "/metrics", "The HTTP path where prometheus metrics will be exposed. Default is `/metrics`.")
+	logFormatJSON  = flag.Bool("log-format-json", false, "set log formatter to json")
```
Comment (mauriciopoppe) [inline cmd/livenessprobe/main.go:51]: we should follow the same flag as the kubelet which is `logging-format` https://github.com/kubernetes/component-base/blob/v0.25.3/logs/api/v1/options.go#L185

File: cmd/livenessprobe/main.go (hunk 1 above)
Comment (Kartik494) [reply cmd/livenessprobe/main.go:140]: @mauriciopoppe ok i will re check this, if it's possible could you provide me pointer on this.

File: cmd/livenessprobe/main.go (hunk 2 above)
Comment (Kartik494) [reply cmd/livenessprobe/main.go:51]: Sure i will modify this to `logging-format`

File: cmd/livenessprobe/main.go (hunk 1 above)
Comment (mauriciopoppe) [reply cmd/livenessprobe/main.go:140]: I wrote some pointers in https://github.com/kubernetes-csi/node-driver-registrar/issues/228#issuecomment-1281453724

Comment (Kartik494) [reply cmd/livenessprobe/main.go:140]: sure i will check that thanks

**Comment by Kartik494** [comment]
Resolved the merge conflict.

**Comment by Kartik494** [comment]
Resolve the merge conflict.

**Review by ggriffiths — CHANGES_REQUESTED** [review]
Looks like this needs a rebase, feel free to re-request review when this is ready.

**Comment by Kartik494** [comment]
Resolved the merge conflict

**Comment by carlory** [comment]
> [CLA Missing ID](https://api.easycla.lfx.linuxfoundation.org/v2/repository-provider/github/sign/20677718/125428840/168/#/?version=2) [CLA Not Signed](https://api.easycla.lfx.linuxfoundation.org/v2/repository-provider/github/sign/20677718/125428840/168/#/?version=2)
> 
> * ✅ login: Kartik494 / name: Kartik Sharma  ([f7a9dea](https://github.com/kubernetes-csi/livenessprobe/commit/f7a9dea7a52e4936736ea4880d06b762e44e7916))
> * ❌ The commit ([dffae97](https://github.com/kubernetes-csi/livenessprobe/commit/dffae9714b28491909e612900711adb1b7e1873e), [cc85cbe](https://github.com/kubernetes-csi/livenessprobe/commit/cc85cbe089629f16b1e0559a3facc5d1f5f52371), [817e991](https://github.com/kubernetes-csi/livenessprobe/commit/817e9918925f18ed26950e1eae3595a70f3ee1e6), [4ccdb51](https://github.com/kubernetes-csi/livenessprobe/commit/4ccdb51ed06590b738dadd840cef0a0b8ddf4338)). This user is missing the User's ID, preventing the EasyCLA check. [Consult GitHub Help](https://help.github.com/en/github/committing-changes-to-your-project/why-are-my-commits-linked-to-the-wrong-user) to resolve.For further assistance with EasyCLA, [please submit a support request ticket](https://jira.linuxfoundation.org/servicedesk/customer/portal/4).

@Kartik494 you can squash those commits into one to fix it.

```
git rebase -i HEAD~5
```

for example:

**Comment by jsafrane** [comment]
/close
in favor of https://github.com/kubernetes-csi/livenessprobe/pull/202

**Pull request closed without merging**